### ISSUE TITLE
Minimalistic production deployment setup / config

### ISSUE THREAD

**Issue by nickvcb** [ing-bank/baker#252]
What is the recommended `Akka` cluster setup for deployment `Baker` in production if scalability was not an immediate requirement?

For discussion with @nikolakasev my understanding is

1. For journal used to store the process events have choice between : in-memory or with a backing store like Cassandra

2. Cluster node count: at least three (majority rule)
> "A single node is never a good idea, and you’d want at least three (majority rule)."

**Comment by FrancoAra** [comment]
In my opinion, the main 2 reasons you want a cluster is for scalability and resilience, if you don't require scalability (though you would almost get it by default because you only need to add new nodes) but you still require resilence (if nodes go down the recipe instances self-heal by respawning on another node) then you still need a backing store like Cassandra, otherwise recipe instance "rehydration" cannot happen.

If you don't need that kind of resilience even, then I would recommend just running in 1 machine with the default local configuration, so no Akka cluster and only local storage., the same configuration that you probably have for your tests.

**Comment by FrancoAra** [comment]
The Baker 3.0 documentation page (which we are working on right now) will have more about this :)

**Comment by SemanticBeeng** [comment]
> no Akka cluster and only local storage

In the current platform am building we rely on `Baker` (`Akka` persistence) for distributed process state management : 
 * resuming a execution of a process from any node in the deployment
 * reading `Akka` persistence journal from any node, 
 etc.

For that we need cluster setup, correct?

**Comment by FrancoAra** [comment]
You are completely correct. Hence the recommendation is to configure journaling to use Cassandra. It is the store we've used for production systems and have worked great
